### PR TITLE
KK-49 | Fetch & name from profile for registration form

### PR DESCRIPTION
**Pull request by badrange**

### CODE REVIEW

File: src/domain/home/form/HomePreliminaryForm.tsx [hunk 1]
```diff
@@ -42,17 +42,24 @@ const HomePreliminaryForm: FunctionComponent<Props> = ({
...
           birthdate: formatTime(
             newMoment(
               `${values.child.birthdate.year}-${values.child.birthdate.month}-${values.child.birthdate.day}`,
               'YYYY-MM-DD'
             )
           ),
-          // Ensure date that saved in redux store was using backend time format.
           firstName: stateFormValues.child.firstName,
-          lastName: stateFormValues.child.firstName,
+          lastName: stateFormValues.child.lastName,
```
Comment (badrange) [inline src/domain/home/form/HomePreliminaryForm.tsx]: This fixes a funny bug

File: src/domain/home/form/HomePreliminaryForm.tsx [hunk 2]
```diff
@@ -42,17 +42,24 @@ const HomePreliminaryForm: FunctionComponent<Props> = ({
...
           homeCity: values.child.homeCity,
         },
+        guardian: {
+          firstName: stateFormValues.guardian.firstName,
```
Comment (midori0507) [inline src/domain/home/form/HomePreliminaryForm.tsx]: You sure this is needed? It suppose to still work without changes on guardian

File: src/domain/registration/form/RegistrationForm.tsx [hunk 3]
```diff
@@ -13,22 +14,46 @@ import { setFormValues } from '../state/RegistrationActions';
...
+  };
+  if (tunnistamoUserValues) {
+    tunnistamoProfile = {
+      familyName: tunnistamoUserValues.profile.family_name,
+      givenName: tunnistamoUserValues.profile.given_name,
+      email: tunnistamoUserValues.profile.email,
+    };
+  }
+
+  tunnistamoUserValues
```
Comment (midori0507) [inline src/domain/registration/form/RegistrationForm.tsx]: This is bad, no console

File: src/domain/registration/form/RegistrationForm.tsx [hunk 4]
```diff
@@ -13,22 +14,46 @@ import { setFormValues } from '../state/RegistrationActions';
...
   // TODO: Do something with the data we get from the backend.
   const [submitChild] = useMutation(submitChildMutationQuery);
   const { t } = useTranslation();
 
+  let tunnistamoProfile: TunnistamoProfile = {
+    familyName: '',
+    givenName: '',
+    email: '',
+  };
+  if (tunnistamoUserValues) {
```
Comment (midori0507) [inline src/domain/registration/form/RegistrationForm.tsx]: You dont need this

File: src/domain/registration/form/RegistrationForm.tsx [hunk 5]
```diff
@@ -13,22 +14,46 @@ import { setFormValues } from '../state/RegistrationActions';
...
 const RegistrationForm: FunctionComponent<Props> = ({
   setFormValues,
   initialValues,
+  tunnistamoUserValues,
 }) => {
   // TODO: Do something with the data we get from the backend.
   const [submitChild] = useMutation(submitChildMutationQuery);
   const { t } = useTranslation();
 
+  let tunnistamoProfile: TunnistamoProfile = {
```
Comment (midori0507) [inline src/domain/registration/form/RegistrationForm.tsx]: You dont need it

File: src/domain/registration/form/RegistrationForm.tsx [hunk 6]
```diff
@@ -114,8 +113,9 @@ const RegistrationForm: FunctionComponent<Props> = ({
                 name="guardian.email"
                 label={t('registration.form.guardian.email.input.label')}
                 onChange={handleChange}
-                value={values.guardian.email}
+                value={values.guardian.email || tunnistamoProfile.email}
```
Comment (midori0507) [inline src/domain/registration/form/RegistrationForm.tsx]: Fill user data in here, no need extra type

File: src/domain/home/form/HomePreliminaryForm.tsx (hunk 2 above)
Comment (badrange) [reply src/domain/home/form/HomePreliminaryForm.tsx]: If you change guardian info in the registration form, and then go back to the front page form, and then back again your own changes will reverted back to what tunnistamo has

File: src/domain/registration/form/RegistrationForm.tsx (hunk 6 above)
Comment (badrange) [reply src/domain/registration/form/RegistrationForm.tsx]: but how are we able to know what the user provided if they want to replace the info from tunnistamo?

File: src/domain/registration/form/RegistrationForm.tsx (hunk 3 above)
Comment (badrange) [reply src/domain/registration/form/RegistrationForm.tsx]: hahaha yeah!

File: src/domain/registration/form/RegistrationForm.tsx (hunk 6 above)
Comment (badrange) [reply src/domain/registration/form/RegistrationForm.tsx]: Hm, I guess we can use tunnistamoProfile.email here since it is not editable

Comment (badrange) [reply src/domain/registration/form/RegistrationForm.tsx]: and lessee

File: src/domain/home/form/HomePreliminaryForm.tsx (hunk 2 above)
Comment (midori0507) [reply src/domain/home/form/HomePreliminaryForm.tsx]: Then we have to find another way to update the state, those changes are non-necessary too long

Comment (badrange) [reply src/domain/home/form/HomePreliminaryForm.tsx]: Any suggestions?

Comment (midori0507) [reply src/domain/home/form/HomePreliminaryForm.tsx]: Done

**Pull request merged**